### PR TITLE
postgresql_14: init at beta 1

### PR DESCRIPTION
**Pull request by marsam**
###### Motivation for this change
https://www.postgresql.org/about/news/postgresql-14-beta-1-released-2213/
https://www.postgresql.org/docs/14/release-14.html

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 125456` at 285fa468 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>postgresql_14</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/stale-substitute.md">stale-substitute</a>

Stale substituteInPlace detected.
substituteStream(): WARNING: pattern '/nix/store/35pnk5kwi26m3ph2bc7dxwjnavpzl8cn-gcc-wrapper-10.3.0/bin/ld' doesn't match anything in file '/nix/store/wqx1qpw4y754rkc1mp2dqbg33ar322ni-postgresql-14beta1/lib/pgxs/src/Makefile.global'
Near pkgs/servers/sql/postgresql/default.nix:158:0:
```
    |
158 |       description = "A powerful, open source object-relational database system";
    | ^
```
</li>
  </ul>
</details>

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -19636,6 +19636,7 @@ in
     postgresql_11
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:19636]: Maybe we want to drop some older version(s)?

Comment (risicle) [reply pkgs/top-level/all-packages.nix:19636]: All versions back to 9.6 inclusive are still supported upstream, so they cost us almost nothing to maintain. And a lot of people run production systems on older db releases as it's hard to schedule the downtime for an upgrade.

**Comment by risicle** [comment]
Builds, macos 10.15

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (marsam) [reply pkgs/top-level/all-packages.nix:19636]: @risicle is right. 
These versions are still widely used in production, we can't drop them so lightly

**Pull request merged**